src/structures/GuildMember.js

```diff
@@ -66,17 +66,19 @@ class GuildMember extends Base {
      */
     this.deleted = false;
 
-    this._roles = [];
-    if (data) this._patch(data);
-  }
-
-  _patch(data) {
     /**
      * The nickname of this member, if they have one
      * @type {?string}
      * @name GuildMember#nickname
      */
-    if (typeof data.nick !== 'undefined') this.nickname = data.nick;
+    this.nickname = null;
+
+    this._roles = [];
+    if (data) this._patch(data);
+  }
+
+  _patch(data) {
+    if (typeof data.nick === 'string') this.nickname = data.nick;
```

Comment:
This has a problem.

If the GuildMember had a nickname set, and then removes it, this if branch won't be called and so it won't be set to null.

Please revert it to the previous condition, `typeof data.nick !== 'undefined'` is more accurate.

Reply:
Oh right, I literally forgot about when the member would edit their nickname, will fix it.